spm/spm_server.c

```diff
@@ -505,16 +513,25 @@ void psa_notify(int32_t partition_id)
         );
     }
 
-    int32_t flags = (int32_t)osThreadFlagsSet(target_partition->thread_id, PSA_SIG_DOORBELL_MSK);
-    SPM_ASSERT(flags >= 0);
+    uint32_t flags = osThreadFlagsSet(target_partition->thread_id, PSA_DOORBELL);
+    // osThreadFlagsSet() sets the msb on failure
+    // flags is allowed to be 0 since by the time osThreadFlagsSet() returns
+    // the flag could have been cleared by another thread
+    SPM_ASSERT((flags & 0x80000000) == 0);
```

Comment:
Is it possible to define 0x80000000 to a macro with the name of bit/flag you are looking, makes it more readable.